core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -630,60 +655,65 @@ trait WorkspaceComponent {
         allGroups <- rawlsGroupQuery.listParentGroupsRecursive(groupRecs.toSet, groupRecs.toSet)
         workspaceRecsForGroups <- findWorkspacesForGroups(allGroups).result
-        workspaceRecsForRealms <- findWorkspacesForRealms(allGroups).result
-        realmedWorkspaceRecs = workspaceRecsForGroups ++ workspaceRecsForRealms
-        accessGroupRecs <- workspaceAccessQuery.filter(rec => rec.workspaceId.inSetBind(realmedWorkspaceRecs.map(_.id))).result
+        workspaceRecsForAuthDomains <- findWorkspacesForAuthDomains(allGroups).result
+        authDomainedWorkspaceRecs = workspaceRecsForGroups ++ workspaceRecsForAuthDomains
+        accessGroupRecs <- workspaceAccessQuery.filter(rec => rec.workspaceId.inSetBind(authDomainedWorkspaceRecs.map(_.id))).result
       } yield {
-        val indexedAccessGroups = accessGroupRecs.map(rec => (rec.workspaceId, rec.accessLevel, rec.isRealmAcl) -> RawlsGroupRef(RawlsGroupName(rec.groupName))).toMap
+        val indexedAccessGroups = accessGroupRecs.map(rec => (rec.workspaceId, rec.accessLevel, rec.isAuthDomainAcl) -> RawlsGroupRef(RawlsGroupName(rec.groupName))).toMap
 
         val groupsToIntersect = for {
-          workspaceRec <- realmedWorkspaceRecs
+          workspaceRec <- authDomainedWorkspaceRecs
           // the if clause below makes sure if the group in question is an access group we don't do the intersection
           // for other access groups of the workspace. The intersections are not required because an access group should
           // not be a sub group of another access group for the same workspace. But more importantly this prevents a
           // concurrency issue. Without this guard changing workspace acls can cause multiple concurrent intersections
           // for all the access groups of a workspace.
           accessLevel <- WorkspaceAccessLevels.groupAccessLevelsAscending if workspaceAccessRec.isEmpty || workspaceAccessRec.get.workspaceId != workspaceRec.id || (workspaceRec.id == workspaceAccessRec.get.workspaceId && accessLevel == WorkspaceAccessLevels.withName(workspaceAccessRec.get.accessLevel))
         } yield {
-          GroupsToIntersect(
-            indexedAccessGroups((workspaceRec.id, accessLevel.toString, true)),
-            indexedAccessGroups((workspaceRec.id, accessLevel.toString, false)),
-            RawlsGroupRef(RawlsGroupName(workspaceRec.realmGroupName.get)))
+          workspaceAuthDomainQuery.filter(_.workspaceId === workspaceRec.id).result.map { authDomainRecs =>
+            val authDomainRefs = authDomainRecs.map(rec => RawlsGroupRef(RawlsGroupName(rec.groupName))).toSet
+            GroupsToIntersect(
+              indexedAccessGroups((workspaceRec.id, accessLevel.toString, true)),
+              authDomainRefs ++ Set(indexedAccessGroups((workspaceRec.id, accessLevel.toString, false)))
+            )
+          }
         }
-        groupsToIntersect.toSet
+        DBIO.sequence(groupsToIntersect)
```

Comment:
yes. the current code does an extra query for every workspace to get all the other groups of the intersection. Instead you can do a single query for all of them at once in the outer for comprehension. Probably something similar to how accessGroupRecs are loaded.

I would strive to have all your queries done in the outer for comprehension.